lib/_stream_readable.js

```diff
@@ -1115,7 +1115,9 @@ async function* createAsyncIterator(stream) {
     destroyImpl.destroyer(stream, err);
     throw err;
   } finally {
-    destroyImpl.destroyer(stream, null);
+    if (state.autoDestroy) {
+      destroyImpl.destroyer(stream, null);
+    }
```

Comment:
I do not think this is correct. We need to call `.destroy` if the `for await` loop is abruptly ended with a `break`. This won't do it unless `autoDestroy` is set.

(Note that the previous behavior was doing that, so we should be ok)

Reply:
So what do you suggest? Something like?

```js
    if (state.autoDestroy || !state.ended) {
      destroyImpl.destroyer(stream, null);
    }
```

Reply:
Yes exactly.

Reply:
Fixed. PTAL